docker/Dockerfile

```diff
@@ -0,0 +1,294 @@
+# has permissions, e.g by being a member of the video group)
+
+
+# TODO: make a custom build of the base image to test with other base OS
+# (e.g ubuntu2204)
+# ???: actually not sure if it's needed or interesting to use the sycl nightly
+# binaries at all. The nightly also ship an up-to-date installation of the
+# intel runtime, including drivers and opencl and level_zero runtimes.
+
+ARG BASE=ghcr.io/intel/llvm/sycl_ubuntu2004_nightly
+ARG TMPDIR=/tmp
+
+# Various resources about the SYCL toolchain
+# Environment variables with important effects on the SYCL toolchain:
+# https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md#sycl_device_filter
+# Overview of the SYCL build instructions and environment requirements: https://intel.github.io/llvm-docs/GetStartedGuide.html
+# More about sycl docker image Dockerfiles: https://github.com/intel/llvm/tree/sycl/devops/containers
+# and https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/DockerBKMs.md
+
+
+ARG INTEL_PYPI_URL=https://pypi.anaconda.org/intel/simple
+ARG BASE_PYPI_URL=https://pypi.org/simple
+
+# /!\ the following version must exist in intel pypi. It is recommended to
+# not use version ranges, because if a higher version exists in pypi.org it
+# will be preferred over the versions in the intel pypi.
+
+ARG INTEL_NUMPY_VERSION="==1.22.3"
+
+# TODO: This can be fixed if intel can host the whole the dependency tree
+# in the INTEL_PYPI_URL instead of having to rely in pypi.org for the main part
+# of the dependency tree.
+# TODO: find which part of the intel basekit should be installed
+# with the installer, and which part can be used from python packages, at
+# build time and at run time, might enable a lighter build.
+
+# ???: not clear if those versions should also be limited to the content
+# of the intel pypi or not. (looks like not)
+ARG INTEL_SCIPY_VERSION=""
+ARG INTEL_NUMBA_VERSION="<0.56"
+
+
+# So far, the intel numpy python package is not built for python>=3.10
+ARG PYTHON_VERSION=3.9.13
+
+
+# This link might need to be refreshed occasionally.
+# It can be found at https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html?operatingsystem=linux&distributions=webdownload&options=online
+# The installer CLI is documented here: https://www.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-linux/top/installation/install-with-command-line.html#install-with-command-line_interactive
+
+ARG ONEAPI_INSTALLER=https://registrationcenter-download.intel.com/akdlm/IRC_NAS/18673/l_BaseKit_p_2022.2.0.262.sh
+ARG ONEAPI_INSTALL_DIR=/opt/intel/oneapi
+
+
+# Some build steps requires a high enough version of cmake.
+# Bump it if necessary.
+
+ARG CMAKE_VERSION=3.23
+ARG CMAKE_VERSION_BUILD=2
+
+
+# Versions of the intel python packages
+
+ARG DPCTL_GIT_BRANCH=0.13.0dev2
+ARG DPCTL_GIT_URL=https://github.com/IntelPython/dpctl.git
+
+ARG DPNP_GIT_BRANCH=master
```

Comment:
The `master` branch potentially might change between image builds.
Must we use tags or even commits instead to reference specific versions and hence reproducible builds?

If so, we might want to enforce this for other projects' branches as well.